doc/man/config.rst

```diff
@@ -0,0 +1,49 @@
+Config file support
+===================
+
+Config files are supported by most subcommands of ``pkgdev`` from any of three
+locations. Listed in order of increasing precedence these include the
+following:
+
+- system config -- /etc/pkgdev/pkgdev.conf
+- user config -- ~/.config/pkgdev/pkgdev.conf
+- custom config -- specified via the --config option
+
+Any settings from a config file with higher precedence will override matching
+settings from a config file with a lower precedence, e.g. user settings
+override system settings. Note that command line options override any matching
+config file setting.
+
+In terms of file structure, basic INI formatting is required and allows
+creating a default section (DEFAULT) for system-wide settings or repo-specific
+sections. The INI key-value pairs directly relate to the available
+long-options supported by the various prefixed by the subcommand name and their
+related values. To find all possible configuration options, run:
+``pkgdev {subcommand} --help``. See the following examples for config settings:
+
+- Run ``pkgcheck scan`` before committing and asks for confirmation (instead of
+  aborting) when creating commits with QA errors::
+
+    [DEFAULT]
+    commit.scan = true
+    commit.ask =
+
+- Allow pushing commits with QA errors, but only for the 'gentoo' repository::
+
+    [gentoo]
+    push.ask =
```

Comment:
It feels confusing to me that it's called .ask and we set it to.. blank? This means "don't ask, it's ok to push even with errors"?

Why is it not .ask = true?

Maybe we can change it to:
"Allow pushing (ask instead of aborting) commits with QA errors, but only for the 'gentoo' repository::"
if my interpretation is correct?

Reply:
Hm, this is intersting as this is part of the command line arguments itself.
If for example you want to pass `--arg` (no value) you do it using `cmd.arg = `, but if you pass value as `--foo bar` then you use `cmd.foo = bar`.
Maybe we should change the impl of `push.ask` to accept optional boolean?

Reply:
I think optional boolean might be nice?

Reply:
Probably would just require adding `const=False, action=arghparse.StoreBool` to the `--ask` arg setup. See the `--scan` and `--mangle` options for examples that default to true using the same support.

Reply:
Done